nba_core/app.py

```diff
@@ -0,0 +1,68 @@
+import os
+
+from flask import Flask
+
+from dal.database import connect_db
+from endpoints.endpoint_functions import before_request_endpoint, after_request_endpoint, user_signup_endpoint, home_endpoint, login_endpoint, logout_endpoint, get_player_stats_endpoint, get_adv_player_stats_endpoint
+
+app = Flask(__name__)
+
+app.config['SQLALCHEMY_DATABASE_URI'] = os.environ.get(
+    'DATABASE_URL', 'postgresql:///nba-users')
+app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+app.config['SQLALCHEMY_ECHO'] = True
+app.config['SECRET_KEY'] = os.environ.get('SECRET_KEY', 'oneonetwotwo')
+
+connect_db(app)
+
+
+@app.before_request
+def add_user_to_g():
+    """If user is logged in, add curr user to Flask global."""
+    before_request_endpoint()
+
+
+@ app.after_request
+def add_header(req):
+    """Add non-caching headers on every request."""
+    return after_request_endpoint(req)
+
+######################################################################
+# General routes
+
+
+@app.route('/')
+def home():
```

Comment:
Small thing, once you've moved your endpoint funcs off to another module the `@route()` method of setting up routes doesn't make as much sense. Check out [`add_url_rule`](https://flask.palletsprojects.com/en/1.1.x/api/#flask.Flask.add_url_rule) in flask. This is what `@route` calls internally and its a great way to wire things up if your simply pointing to another func.